core/domain/exp_domain.py

```diff
@@ -3694,11 +3760,11 @@ def _convert_v5_dict_to_v6_dict(cls, exploration_dict):
 
         # Ensure this exploration is up-to-date with states schema v3.
         exploration_dict['states'] = cls._convert_states_v0_dict_to_v1_dict(
-            exploration_dict['states'])
+            exploration_dict['states'], exploration_dict['id'])
```

Comment:
OK, wait a sec. Now I don't get it. Why do you have to add this ID field everywhere when it's needed in only one of the methods? Why not add it for just the method in question?

It's probably better to use an if condition (if needed) to check the particular version, rather than making so many changes in unrelated functions in order to pass in params that are not needed. That is kind of a "code smell".

Reply:
Yeah. I have added an if condition for this particular function only.